### PR TITLE
Feature - Begin form front end

### PR DESCRIPTION
**Pull request by christulin**
## Trello Ticket
PR towards the completion of https://trello.com/c/FsH7dPyD/38-contact-form-front-end-design-and-functionality

## How to Setup
`npm i`(if not done already)
`npm run start`
`npm run server`
Open up http://localhost:3001/

## Purpose of Pull Request
The self serve options and logic to pull up both form types should be complete.  I have only begun work on the 'get in touch' form.  Play around with adding and removing text, I have only implemented an error message for first name though.

Any and all feedback is appreciated.. I think my scss files still resemble a bit of spaghetti :)

### CODE REVIEW

File: client/Components/FormContainer.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const FormContainer = () => {
+
+  const [ formType, setFormType ] = useState('self-serve');
+
+  const handleFormChange = (formType) => {
+    setFormType(formType);
+  }
+
+  let form;
```
Comment (christulin) [inline client/Components/FormContainer.js:14]: Open to cleaner ways of handling this conditional rendering logic

File: client/Components/GetInTouchForm.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+
+const GetInTouchForm = ( { handleFormChange }) => {
+
+  const [isHidden, toggleHidden] = useState(true);
```
Comment (christulin) [inline client/Components/GetInTouchForm.js:5]: would like to figure out way to toggle hidden on/off for individual inputs without having multiple `useState` functions

File: client/Styles/form.scss [hunk 3]
```diff
@@ -0,0 +1,144 @@
+.sticky-form-wrap {
```
Comment (christulin) [inline client/Styles/form.scss:1]: Any and all styling criticism is very much appreciated!

Comment (christulin) [reply client/Styles/form.scss:1]: kind of a mixture of my own stuff plus taking exact measurements from wework site

File: client/Components/GetInTouchForm.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    const field = e.target.parentElement;
+
+    //can eventually add other helper functions here to do some *basic* front end validation
+    if (!e.target.value) {
+
+      if (field.classList.contains('complete')) field.classList.remove('complete');
+
+      field.classList.add('error');
+      toggleHidden(false);
```
Comment (christulin) [inline client/Components/GetInTouchForm.js:24]: known bug: this toggles on same error message no matter what field it was triggered by

File: client/Components/FormContainer.js (hunk 1 above)
Comment (samasno) [reply client/Components/FormContainer.js:14]: Maybe you could do the formType check inside each component to clean up the container

File: client/Components/GetInTouchForm.js (hunk 2 above)
Comment (samasno) [reply client/Components/GetInTouchForm.js:5]: try making an object with useState that has a prop for each input status

File: client/Styles/form.scss (hunk 3 above)
Comment (samasno) [reply client/Styles/form.scss:1]: looks pretty good so far but noticed alot of repeated style values. id recommend declaring sass variables and using the extend/inheritance features

**Review by samasno — APPROVED** [review]
Made a few suggestions but otherwise everything works as expected

**Pull request merged**